data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/PluginCreator.java

```diff
@@ -23,7 +23,7 @@
 class PluginCreator {
     private static final Logger LOG = LoggerFactory.getLogger(PluginCreator.class);
 
-    <T> T newPluginInstance(final Class<T> pluginClass,
+    public <T> T newPluginInstance(final Class<T> pluginClass,
```

Comment:
This shouldn't be public. The class itself is already package protected, and keeping the method package protected is clearer.